### ISSUE TITLE
Gitea support at all?

### ISSUE THREAD

**Issue by mannp** [Mic92/buildbot-nix#91]
Hi there

I was hoping for an alternative to Hydra that connects to something other than Github and even to be able to use private repos?

You mention plans to support Gitea, do you still have those plans at some stage?

Thanks.

**Comment by Mic92** [comment]
Yes. I still have plans and three potential places where I could make use of it. Probably not this month but maybe still in Q1/2024.

**Comment by mannp** [comment]
Great news 👌🏻 and thanks 🙂

**Comment by Mic92** [comment]
Work started in https://github.com/Mic92/buildbot-nix/pull/124

**Comment by Mic92** [comment]
Works!! https://git.clan.lol/clan/clan-core/pulls/1306

**Comment by mannp** [comment]
> Works!! https://git.clan.lol/clan/clan-core/pulls/1306

Many thanks @Mic92 :) .. will give it a try.

**Comment by mannp** [comment]
Hey there @Mic92 hope it's okay asking for follow-up questions here :)

I have my instance up and running with worker_preparation and reload_gitea_projects 'green', but;

1) What is the endpoint URL for the webhook in buildbot for Gitea? (I found this for the gitea plugin for normal buildbot but it doesn't appear to work -> https://buildbot.domain/change_hook/gitea/)
2) What is the minimum scope of the api token required -> repo read/write? user read? Thanks.
3) this one feels extra dumb, but I will ask anyway :) What is the format of the worker defn in sops-nix please, as I cannot get it to work with below;

```
    nix-workers: |
        ''
          [
            { "name": "builder", "pass": "password", "cores": 6 }
          ]
        ''
```

Many thanks again for building this :)

**Comment by Mic92** [comment]
I expanded the section a bit: https://github.com/Mic92/buildbot-nix?tab=readme-ov-file#integration-with-gitea

You worker file shouldn't contain the `''` quotes, otherwise it's correct.

**Comment by mannp** [comment]
Thanks for doing it so quickly, I will have a look.

1) Oh I see, I didn't realise, but did check and there were no webhooks created in the buildbot user I created....likely I didn't give the right permissions in the token for that to happen.

I have completely missed this "topic" being set, I will look again at the docks to see where this needs to be.

3) Strange, I did try without the quotes and the master failed to start. I will take another look.
